### ISSUE TITLE
Feature request: Add option of entity or bundle when doing doctrine:schema:create

### ISSUE THREAD

**Issue by MicWit** [doctrine/dbal#2707]
This is a request I actually uploaded to Symfony https://github.com/symfony/symfony/issues/22606 and they said I should request it here. I will copy and paste it, so these commands are actually for the Symfony console (I have only ever used doctrine through Symfony).

I wanted to look at what sql was being run in a bundle I was installing, and so used
`php bin/console doctrine:schema:create --dump-sql`
This came up with a lot of SQL, and in some cases, it would be nice to be able to just create or update one entity or entities under 1 bundle. When I looked into it, there is the option of using one entity manager, but that generally isn't going to help with vendor bundles etc.

So it would be nice to get an option on the console command for schema create or schema update to select just one entity or any entity within a bundle (I understand that the bundle option may be hard to do, but if we could just do it by one, or maybe a number of entities, we could just look in the bundle at the entity names).

So then we could do something like
`php bin/console doctrine:schema:create --dump-sql --entities=[entity1,entity2]`
and narrow down which entities are affected.

**Comment by Ocramius** [comment]
Shouldn't this just be `./cli doctrine:schema:create --dump-sql | grep
'(entity1|entity2)`?

**Comment by MichaelWitten-SCU** [comment]
No, thats just searching for text in the sql dump I believe. This would need to filter out that entit(y/ies) and just do an sql dump if there is a match.

**Comment by MicWit** [comment]
No, thats just searching for text in the sql dump I believe. This would need to filter out that entit(y/ies) and just do an sql dump if there is a match.

**Comment by Ocramius** [comment]
Yes, but since you are asking for the feature for debugging purposes, and
most times you can't dump single entities due to partially covered FK
definitions: what is the advantage over a simple `grep`?

**Comment by MicWit** [comment]
Well a grep searches the text, so you wont get indexes or anything, but there will also be times that you don't want to update the entire database. For example, if you are using external data in views for your site and then you need to install a vendor bundle. You will want to just install their part or you are going to screw your manual entries you had to do for the view (as views do not play happy in doctrine all the time).

There will also be other times that you don't want to create/update all tables etc in the database, especially when coming from an external vendor etc.

**Comment by Ocramius** [comment]
>  you wont get indexes or anything, 

You will? Most statements will be (for example, on MySQL) `ALTER TABLE x ...`, so you can `grep` for `x`

 > There will also be other times that you don't want to create/update all tables etc in the database, especially when coming from an external vendor etc.

That is one of those scenarios that I really wouldn't want to support, because it leads to the same issues as what we had with `EntityManager#flush($somethingSpecific)`: broken graph, missing listener calls, etc. The same applies to the schema (broken graph, missing listener calls, etc.).
